posix: clock: fix integer overflow in __z_clock_nanosleep

tagunil opened the pull request:
Nanosecond time calculation overflows if the libc has 32-bit time_t. One such libc is the classic ARC MWDT one, but there might be others.

cfriedt reviewed (changes requested):
Can this commit be separated into code changes and formatting changes, if necessary? Or if the formatting commit is not necessary, then please just submit only the code changes.

tagunil commented:
> Can this commit be separated into code changes and formatting changes, if necessary? Or if the formatting commit is not necessary, then please just submit only the code changes.

I've just dropped the second commit which was entirely formatting, but I believe the first one should contain some formatting changes to avoid clang-format compliance warnings. If it's not the case, I can split it further.

cfriedt reviewed (approved):
LGTM

ycsin commented on lib/posix/options/clock.c:
```diff
@@ -216,13 +216,11 @@ static int __z_clock_nanosleep(clockid_t clock_id, int flags, const struct times
-		ns = rqtp->tv_nsec + NSEC_PER_SEC
-			+ k_sleep(K_SECONDS(rqtp->tv_sec - 1)) * NSEC_PER_MSEC;
+	if ((flags & TIMER_ABSTIME) == 0 && unlikely(rqtp->tv_sec >= ULLONG_MAX / NSEC_PER_SEC)) {
+		ns = rqtp->tv_nsec + NSEC_PER_SEC +
+		     (uint64_t)k_sleep(K_SECONDS(rqtp->tv_sec - 1)) * NSEC_PER_MSEC;
 	} else {
-		ns = rqtp->tv_sec * NSEC_PER_SEC + rqtp->tv_nsec;
+		ns = (uint64_t)rqtp->tv_sec * NSEC_PER_SEC + rqtp->tv_nsec;
```
I think the first case isn't due to `time_t` being 32 bit? If so could you please update the commit message or split into 2 commits to reflect that?

tagunil replied:
Sure, just done that.

ycsin reviewed (commented):
clang-format warnings shouldn't cause the CI to fail, I also prefer splitting the formatting part into another commit

tagunil committed:
posix: clock: fix 32-bit time_t overflow in __z_clock_nanosleep

Nanosecond time calculation overflows if the libc has 32-bit time_t.
One such libc is the classic ARC MWDT one, but there might be others.

Signed-off-by: Ilya Tagunov <Ilya.Tagunov@synopsys.com>

tagunil committed:
posix: clock: fix potential int32_t overflow in __z_clock_nanosleep

As k_sleep returns int32_t, there is a possibility for integer overflow
during conversion from milliseconds to nanoseconds.

Signed-off-by: Ilya Tagunov <Ilya.Tagunov@synopsys.com>

tagunil committed:
posix: clock: partially reformat __z_clock_nanosleep

Apply clang-format to the offending lines to make it happy.

Signed-off-by: Ilya Tagunov <Ilya.Tagunov@synopsys.com>

ycsin commented:
CI error looks unrelated, prob need a rebase
